.github/workflows/tests.yml

```diff
@@ -305,11 +305,19 @@ jobs:
         with:
           path: synapse
 
-      - name: Run actions/checkout@v2 for complement
-        uses: actions/checkout@v2
-        with:
-          repository: "matrix-org/complement"
-          path: complement
+      # Attempt to check out the same branch of Complement as the PR. If it
+      # doesn't exist, fallback to master.
+      - name: Checkout complement
+        shell: bash
+        run: |
+          # For pull requests we want to use GITHUB_HEAD_REF (the pull request
+          # head branch). Otherwise, we use GITHUB_REF (refs/heads/<branch>).
+          BRANCH_NAME="$GITHUB_HEAD_REF"
```

Comment:
Is `$GITHUB_HEAD_REF` right, here? Should we instead (or also?) look at `$GITHUB_BASE_REF`?

E.g., if I have a branch called `issue-451` and make a PR against Synapse's `release-v1.34` branch, then I'm looking at:

    GITHUB_HEAD_REF="issue-451"
    GITHUB_BASE_REF="release-v1.34"
    GITHUB_REF="refs/pull/10700/merge"

We probably want to look for matching branches in complement for _both_ the `HEAD` and `BASE`, before falling back to `master`, right?

Reply:
During the release process for Synapse, we'll typically create and push a matching branch on Sytest. If we started doing the same for Complement, then falling back to the base ref instead of master (which might be ahead of the release branch at that point) is a good idea.

So I think `GITHUB_HEAD_REF` -> `GITHUB_BASE_REF` -> `master` is a sound idea.

Reply:
Concretely, it may be cleaner to do something like

```bash
# Attempt to check out the most relevant branch of Complement, if it exists
#   GITHUB_HEAD_REF and GITHUB_BASE_REF are only set on pulls; otherwise empty
#   GITHUB_REF is /refs/pull/BRANCH on pushes, but refs/pull/123/merge on pulls
# Falls back to master if no more specific branch is found
for branch in "$GITHUB_HEAD_REF" "$GITHUB_BASE_REF" "${GITHUB_REF#refs/heads/}" "master"; do
    if [ -z "$branch" ] || [[ $branch =~ ^refs/pull/.* ]]; then
        continue
    fi

    git clone --shallow --branch "$branch" https://github.com/matrix-org/complement && break
done
```